sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -241,80 +360,197 @@ to ensure that for their first argument `base` and second argument `exponent`:
+template <class T> complex<T> log10(const complex<T> &);
+/// Compute the square root of complex number x.
+template <class T> complex<T> sqrt(const complex<T> &);
+/// Compute the base-e exponent of complex number x.
+template <class T> complex<T> exp(const complex<T> &);
+
+/// Compute complex number z raised to the power of complex number y.
+template <class T> complex<T> pow(const complex<T> &, const complex<T> &);
+/// Compute complex number z raised to the power of complex number y.
+template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const complex<U> &);
+/// Compute complex number z raised to the power of real number y.
+template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const U &);
+/// Compute real number x raised to the power of complex number y.
+template <class T, class U> complex</*Promoted*/> pow(const T &, const complex<U> &);
+
+/// Compute the inverse hyperbolic sine of complex number x.
+template <class T> complex<T> asinh(const complex<T> &);
+/// Compute the inverse hyperbolic cosine of complex number x.
+template <class T> complex<T> acosh(const complex<T> &);
+/// Compute the inverse hyperbolic tangent of complex number x.
+template <class T> complex<T> atanh(const complex<T> &);
+/// Compute the hyperbolic sine of complex number x.
+template <class T> complex<T> sinh(const complex<T> &);
+/// Compute the hyperbolic cosine of complex number x.
+template <class T> complex<T> cosh(const complex<T> &);
+/// Compute the hyperbolic tangent of complex number x.
+template <class T> complex<T> tanh(const complex<T> &);
+/// Compute the inverse sine of complex number x.
+template <class T> complex<T> asin(const complex<T> &);
+/// Compute the inverse cosine of complex number x.
+template <class T> complex<T> acos(const complex<T> &);
+/// Compute the inverse tangent of complex number x.
+template <class T> complex<T> atan(const complex<T> &);
+/// Compute the sine of complex number x.
+template <class T> complex<T> sin(const complex<T> &);
+/// Compute the cosine of complex number x.
+template <class T> complex<T> cos(const complex<T> &);
+// Compute the tangent of complex number x.
+template <class T> complex<T> tan(const complex<T> &);
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+=== Element-Wise Mathematical operations
+
+In harmony with the complex scalar operations, this proposal extends
+furthermore the `sycl::ext::oneapi::experimental`` namespace math functions
+to accept `sycl::marray<complex<T>>` for a range of SYCL math functions.
+
+Specifically, it adds support for `abs`, `acos`, `asin`, `atan`, `acosh`,
+`asinh`, `atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`, `norm`,
+`polar`, `pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+
+Additionally, this extension introduces support for the `real` and `imag` free
+functions, which return marrays of scalar values representing the real and
+imaginary components, respectively.
```

Comment:
Same comment as above.

Reply:
Don't you think that the declarations of these functions shown after this block (with comments) are not enough?

```c++
/// Returns an marray of real components from the marray x.                                                                                                                                                                                                                               
template <typename T, std::size_t NumElements>                                                                                                                                                                                                                                            
sycl::marray<T, NumElements> real(const marray<complex<T>, NumElements> &x);                                                                                                                                                                                                              
/// Returns an marray of imaginary components from the marray x.                                                                                                                                                                                                                          
template <typename T, std::size_t NumElements>                                                                                                                                                                                                                                            
sycl::marray<T, NumElements> imag(const marray<complex<T>, NumElements> &x);
```

Reply:
We don't usually specify behavior in comments like this, so I don't think a reader would think to check them if they were looking for a description of what the functions do.  They're also not a complete description, because they don't explain which values of `T` are supported.

I can see that the comment style was being used by this specification before you modified it, so I'm happy if you want to leave the formatting updates for a later PR.

Reply:
In the `Complex Class` section, it is specified that "The `T` template parameter must be one of the types float, double, or sycl::half."

A possible solution could be to add a note that links to this part of the `Complex Class`.
What do you think?

Reply:
I'm not talking about the overloads that accept a `complex<T>`, but the cases where you don't accept a complex number:

```c++
template <class T> constexpr T real(T);
template <class T> constexpr T imag(T);
...
// and so on
```

Presumably these are intended to support the same types as the ISO C++ version, but I don't see that said anywhere.  Also note that ISO C++ doesn't define these functions this way, but rather as `real(float)`, `real(double)` and `real(long double)`.

Reply:
We shouldn't concern ourselves too much with what cppreference says, because it's just an online reference guide.  It's more important in my opinion to align with the ISO C++ specification, which defines these additional overloads [here](https://eel.is/c++draft/cmplx.over).

Rather than duplicate all of this text, though, the simplest thing to do here would probably just be to add a note explaining that these functions aren't really new, and that their behavior matches ISO C++.

But as you say, this isn't really related to `marray`.  I'm happy for it to be addressed in a later PR.

Reply:
I'll create a PR on top of this branch and will ping you to review the new note added to the `real` and `imag` functions.

I will let you `Resolve the conversation` as a sign of consensus and closure on that topic.